Sources/GRPCHTTP2TransportNIOPosix/HTTP2ServerTransport+Posix.swift

```diff
@@ -211,64 +212,79 @@ extension HTTP2ServerTransport {
       }
 
       try await serverChannel.executeThenClose { inbound in
-        try await withThrowingDiscardingTaskGroup { serverTaskGroup in
+        try await withThrowingDiscardingTaskGroup { group in
           for try await (connectionChannel, streamMultiplexer) in inbound {
-            serverTaskGroup.addTask {
-              try await connectionChannel
-                .executeThenClose { connectionInbound, connectionOutbound in
-                  await withDiscardingTaskGroup { connectionTaskGroup in
-                    connectionTaskGroup.addTask {
-                      do {
-                        for try await _ in connectionInbound {}
-                      } catch {
-                        // We don't want to close the channel if one connection throws.
-                        return
-                      }
-                    }
-
-                    connectionTaskGroup.addTask {
-                      await withDiscardingTaskGroup { streamTaskGroup in
```

Comment:
FWIW this is the unnecessary task group: it only ever has one task so we can just use the connection task group